[adhigop13] iP

adhigop13 opened the pull request:
# Tom

Tom frees your mind of having to remember things you need to do. This program is: 

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use!

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/344). ✔️ 
2. double-click it. 🖱️ 
3. add your tasks. ➕ 
4. let it manage your tasks for you 😄 

And it is **FREE!** 😱 

Features:

- [x] Managing tasks
- [x]   Managing deadlines
- [ ]   Reminders (coming soon)
- [ ]   An awesome GUI (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main `method:

 ```Java
public class Main {
     public static void main(String[] args) {
         Application.launch(MainApp.class, args);
     }
 }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

adhigop13 committed:
Completed Level 0 skeletal version.

adhigop13 committed:
Added echo function lvl1

adhigop13 committed:
Added memory and indentation, lvl-2

adhigop13 committed:
Created new Task class to encapsulate "Task" functionality. Implemented mark and unmark functions.

adhigop13 committed:
Added new files for encapsulating different tasks. Lvl-4 update

adhigop13 committed:
Significant changes made to code structure in the child classes so that input will be taken as one. (eg todo swim). Removes necessity for multiple scanners and passes testcases.

adhigop13 committed:
Added DukeException class, Lvl 5 completed

adhigop13 committed:
Added Delete function, more exceptions to handle errors. Level 6 update

adhigop13 committed:
Replaced old list implementation with enum. Changes in code related to this. Added comments to ensure better code readability...

adhigop13 committed:
Add duke.txt and save function

adhigop13 committed:
Merge branch 'branch-Level-7'

adhigop13 committed:
Add LocalDateTime storage lvl8

adhigop13 committed:
Merge branch 'branch-Level-8'

adhigop13 committed:
Add more OOP, create new classes

adhigop13 committed:
Merge branch 'branch-A-MoreOOP'

adhigop13 committed:
Create packages : A-Packages

adhigop13 committed:
Merge branch 'add-gradle-support'

adhigop13 committed:
Add build.gradle

adhigop13 committed:
Add JUnit Tests for Parser and TaskList

adhigop13 committed:
Create Jar file

adhigop13 committed:
Add JavaDoc comments

adhigop13 committed:
Fix code to follow Java Coding Standard

adhigop13 committed:
Add find function

adhigop13 committed:
Merge branch 'branch-A-JavaDoc'

adhigop13 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/task/Deadline.java
#	src/main/java/task/Event.java
#	src/main/java/task/Task.java
#	src/main/java/task/ToDo.java

adhigop13 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Parser.java
#	src/main/java/dukeUiElements/Ui.java

tyouwei commented on data/duke.txt:
add this file to .gitignore

tyouwei reviewed (changes requested):
LGTM! Just some nits to fix.

adhigop13 committed:
Add basic GUI(Pre FXML)  Level 10

tiongjjyi commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+public class Deadline extends Task {
+    private String deadlineBy;
+    private LocalDateTime deadlineInDateTime;
```
Good naming. It is very intuitive and explains the named entity to the reader accurately and at a sufficient level of detail.

tiongjjyi commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,87 @@
+ */
+public class Task {
+    private String description;
+    private boolean status;
```
Perhaps can name this boolean sounds more like a boolean?

adhigop13 replied:
Oops! Thanks for catching that! Will update on my next commit!

tiongjjyi reviewed (commented):
Overall is perfect, can barely see any violation in naming.

erohsikivar commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,83 @@
+     * @return boolean value false to break out of program, true otherwise.
+     * @throws DukeException when incorrect / invalid input is entered.
+     */
+    public static String parse(String userInput) throws DukeException {
```
Good to see that you have handled all the cases of inputs, you could also try using switch and case instead of if-else.

erohsikivar commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @param pathOfDirectory Path variable provided
+     * @param storeTask ArrayList<Task> which stores tasks when program is running
+     * @throws IOException if an I/O error occurs
```
Good to see that you handled a case whereby the file does not exist.

erohsikivar commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,45 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + printDateTimeFormat(this.fromDateTime) + " to: " + printDateTimeFormat(this.toDateTime) + ")";
```
Please try not to code more than 120 characters on the same according to Java Coding Standard.

erohsikivar reviewed (commented):
Overall looks good to me, you can consider doing some minor changes.

adhigop13 committed:
Add checkstyle mods

adhigop13 committed:
Add assertions

adhigop13 committed:
Add FXML support

adhigop13 committed:
Refactor code for better code quality

Several code were changed from Arrowhead style to more readable
format.

Changed if-else to switch-case-default for better readability.

As a step towards refactoring, several new methods have been created
to improve the readability of code.

adhigop13 committed:
Merge pull request #2 from adhigop13/branch-A-Assertions

Add assertions

adhigop13 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

adhigop13 committed:
Add improved find functionality

adhigop13 committed:
Improve GUI to support chat bubbles

adhigop13 committed:
Fix checkstyle errors

adhigop13 committed:
Add significant changes to Ui and code quality

adhigop13 committed:
Fix Junit assertEquals error

adhigop13 committed:
Add green/blue text bubbles

adhigop13 committed:
Fix minor bugs

adhigop13 committed:
update README.md

adhigop13 committed:
Add Ui.png

adhigop13 committed:
Add minor bug fixes and add images to README.md

adhigop13 committed:
Modify README.md

adhigop13 committed:
Add indentation in README.md

adhigop13 committed:
Add changes from static to instance variables

adhigop13 committed:
Add stage title

adhigop13 committed:
Fix minor bugs in TaskList class

adhigop13 committed:
Add acknowledgement for save method

adhigop13 committed:
Fix compatibility issues on Linux
